src/engine/strat_engine/thinpool/thinpool.rs

```diff
@@ -314,7 +319,7 @@ impl ThinPool {
                     // A more sophisticated approach might be in order.
                     match self.extend_thinpool(dm, usage.total_data, bd_mgr) {
                         #![allow(single_match)]
-                        Ok(_) => {}
+                        Ok(_) => should_save = true,
```

Comment:
If you are going to take this approach for ```extend_thinpool``` call, it also applies to ```extend_thinpool_meta```.

Reply:
very true! ok.